partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -1070,6 +1155,895 @@ class PartiQLSchemaInferencerTests {
+                    )
+                )
+            ),
+            ErrorTestCase(
+                name = "invalid exclude collection wildcard",
+                query = """SELECT * EXCLUDE t.a[*]
+                    FROM <<
+                        {
+                            'a': {
+                                'b': {
+                                    'c': 0,
+                                    'd': 'foo'
+                                }
+                            }
+                        }
+                    >> AS t""",
+                problemHandler = assertProblemExists {
+                    Problem(
+                        UNKNOWN_PROBLEM_LOCATION,
+                        PlanningProblemDetails.InvalidExcludeExpr
+                    )
+                }
+            ),
+            ErrorTestCase(
+                name = "invalid exclude collection index",
+                query = """SELECT * EXCLUDE t.a[1]
+                    FROM <<
+                        {
+                            'a': {
+                                'b': {
+                                    'c': 0,
+                                    'd': 'foo'
+                                }
+                            }
+                        }
+                    >> AS t""",
+                problemHandler = assertProblemExists {
+                    Problem(
+                        UNKNOWN_PROBLEM_LOCATION,
+                        PlanningProblemDetails.InvalidExcludeExpr
+                    )
+                }
+            ),
+            ErrorTestCase(
+                name = "invalid exclude tuple attr",
+                query = """SELECT * EXCLUDE t.a.b
+                    FROM <<
+                        {
+                            'a': [
+                                { 'b': 0 },
+                                { 'b': 1 },
+                                { 'b': 2 }
+                            ]
+                        }
+                    >> AS t""",
+                problemHandler = assertProblemExists {
+                    Problem(
+                        UNKNOWN_PROBLEM_LOCATION,
+                        PlanningProblemDetails.InvalidExcludeExpr
+                    )
+                }
+            ),
+            ErrorTestCase(
+                name = "invalid exclude tuple wildcard",
+                query = """SELECT * EXCLUDE t.a.*
+                    FROM <<
+                        {
+                            'a': [
+                                { 'b': 0 },
+                                { 'b': 1 },
+                                { 'b': 2 }
+                            ]
+                        }
+                    >> AS t""",
+                problemHandler = assertProblemExists {
+                    Problem(
+                        UNKNOWN_PROBLEM_LOCATION,
+                        PlanningProblemDetails.InvalidExcludeExpr
+                    )
+                }
+            ),
```

Comment:
I'm curious about the behavior when an EXCLUDE attribute is removed and referenced later on in the exclude list. Something like:
```
SELECT * EXCLUDE t.a, t.a.b FROM << { 'a': { 'b': 1 } } >> AS t 
```